File: Benchmark/erathostene.jl
using PythonCall
function cribble_erathostene(n::Int64=10)
    list_boolean_prime = ones(Bool,n)

    for i in range(2,n)
        for j in range(i+1,n)
            # Si j est potentiellement premier
            if (list_boolean_prime[j])
                # Si i divise j
                if (rem(j,i) == 0)
                    # j n'est pas premier
                    list_boolean_prime[j] = false
                end
            end
        end
    end
    # On retourne les index de list_boolean_prime dont la valeur est true.
    index_primes = findall(list_boolean_prime)
    print("Les nombres premiers de 1 à " * string(n) * " sont :")
    print(index_primes)
    return index_primes
end

@time cribble_erathostene(10000)

File: Benchmark/erathostene.py
import time
def cribble_erathostene(n=10):
    liste_boolean_prime = [True for _ in range(0,n)]
    for i in range(1,n):
        for j in range(i+1,n):
            if liste_boolean_prime[j]:
                if divmod(j+1,i+1)[1] == 0:
                    liste_boolean_prime[j] = False
        
    index_primes = [i+1 for i, x in enumerate(liste_boolean_prime) if x]
    print("Les nombres premiers de 1 à " + str(n) + " sont :")
    print(index_primes)
    return index_primes

number= 10000
time_start = time.time()
cribble_erathostene(n=number)
time_final = time.time() - time_start
print("For " + str(number) + ", it took : " + str(time_final) + 'seconds.')